RPIPs/RPIP-draft.md

```diff
@@ -0,0 +1,95 @@
+indices submitBalances right away but only submitPrices after waiting X minutes. On an odd update,
+nodes with odd indices submitBalances right away but only submitPrices after waiting 10 minutes.
+Nodes should do the work immediately and hold off on voting, to avoid unnecessary delay when
+submitting after their 10-minute wait. Note: consensus currently requires more than half of votes,
+so this approach will often mean we need to hit the 10-minute wait before consensus is possible.
+
+### Subcommittee for reward snapshot duty
+This change can be implemented now; however, trees currently take a long time and the subcommittee
+approach works using generous buffers. As a result, it's recommended that this change be implemented
+after trees are reliably fast enough in the watchtower such that this approach doesn't degrade NO
+experience significantly.
+
+Each oDAO node figures out its own index (similar code already exists). On an even rewards period,
+nodes with even indices submitBalances right away but only submitPrices after waiting 30 minutes.
+On an odd update, nodes with odd indices submitBalances right away but only submitPrices after
+waiting X minutes. Nodes should do the work immediately and hold off on voting, to avoid unnecessary
+delay when submitting after their 30-minute wait. Note: consensus currently requires more than half
+of votes, so this approach will often mean we need to hit the 10-minute wait before consensus is
+possible.
+
+It should be explicitly noted that isolated failures are not cause for alarm, and there are simply
+not many data points. The oDAO may wish to conduct an internal test with a member that didn't submit
+when in the active subcommittee.
+
+### Voting past consensus
+This change requires a smart contract release.
+
+It's near-certain that there is a more gas-optimal intervention here; in particular, some conditions
+are nearly-identically checked twice when hitting the execution threshold.
+
+#### submitPrices()
+- Change > to >= in `require(_block > getPricesBlock(), "Network prices for an equal or higher block are set");`
+  - Update message to say "for a higher block"
+- Change `updatePrices(_block, _rplPrice);` to `if (_block != getPricesBlock()) {updatePrices(_block, _rplPrice);}`
+
+#### submitBalances()
+- Change > to >= in `require(_block > getBalancesBlock(), "Network balances for an equal or higher block are set");`
+  - Update message to say "for a higher block"
+- Change `updateBalances(_block, _totalEth, _stakingEth, _rethSupply);` to `if (_block != getBalancesBlock()) {updateBalances(_block, _totalEth, _stakingEth, _rethSupply);}`
+
+### submitRewardSnapshot()
+- Change == to <= in `require(_submission.rewardIndex == getRewardIndex(), "Can only submit snapshot for next period");`
+  - fix message to "Can only submit snapshot for periods up to next" 
+- Change > to >= `require(_submission.intervalsPassed > 0, "Invalid number of intervals passed");`
```

Comment:
This line should remain as is, `intervalsPassed` is part of the snapshot submission it doesn't prevent someone from submitting a past vote.